python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -48,6 +50,96 @@
     from dagster._utils.caching_instance_queryer import CachingInstanceQueryer  # expensive import
 
 
+class _AutoMaterializeCondition(Enum):
+    FRESHNESS = "FRESHNESS"
+    DOWNSTREAM_FRESHNESS = "DOWNSTREAM_FRESHNESS"
+    PARENT_UPDATED = "PARENT_UPDATED"
+    MISSING = "MISSING"
+
+
+@whitelist_for_serdes
+class AutoMaterializeReason(NamedTuple):
+    condition: _AutoMaterializeCondition
+
+    @staticmethod
+    def freshness() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.FRESHNESS)
+
+    @staticmethod
+    def downstream_freshness() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.DOWNSTREAM_FRESHNESS)
+
+    @staticmethod
+    def parent_updated() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.PARENT_UPDATED)
+
+    @staticmethod
+    def missing() -> "AutoMaterializeReason":
+        return AutoMaterializeReason(condition=_AutoMaterializeCondition.MISSING)
+
+
+class _AutoMaterializeSkipCondition(Enum):
+    PARENT_UNRECONCILED = "PARENT_UNRECONCILED"
```

Comment:
I think we should avoid the word "RECONCILE"

Reply:
agree -- i think this should be split out into two things, `PARENT_MISSING` and something else, denoting that the parent has new data available that it has not yet incorporated. This one is harder to name... some attempts:

`PARENT_NOT_READY`
`PARENT_NOT_UP_TO_DATE`
`PARENT_INCONSISTENT`
`PARENT_NEEDS_UPDATE`
`PARENT_STALE` (lol)
`PARENT_OUTDATED`

I went with `PARENT_OUTDATED` for now, and am punting on disambiguating between that and `PARENT_MISSING` until `is_reconciled` is updated to provide more specific reasons